packages/ai/src/lib/TldrawAiModule.ts

```diff
@@ -32,141 +25,87 @@ export class TldrawAiModule {
 	 * Creates and prepare a prompt, returning the prompt
 	 * and a function to handle changes.
 	 *
-	 * @param prompt - The user's message or a configuration for the prompt
+	 * @param options - The user's message or a configuration for the prompt
 	 */
-	async generate(prompt: string | { message: TLAiMessages; stream?: boolean }) {
+	async generate(options: TldrawAiPromptOptions) {
 		const { transforms: _transformCtors = [] } = this.opts
 		const transforms = _transformCtors.map((ctor) => new ctor(this.opts.editor))
 
-		const message = typeof prompt === 'string' ? prompt : prompt.message
-		let _prompt = await this.getPrompt(message)
+		let prompt = await this.getPrompt(options)
 
 		for (const transform of transforms) {
 			if (transform.transformPrompt) {
-				_prompt = transform.transformPrompt(_prompt)
+				prompt = transform.transformPrompt(prompt)
 			}
 		}
 
 		transforms.reverse()
 
-		const handleChange = (change: TLAiChange) => {
+		const handleChange = (change: TLAiChange, apply: TldrawAiApplyFn) => {
 			for (const transform of transforms) {
 				if (transform.transformChange) {
 					change = transform.transformChange(change)
 				}
 			}
-			this.applyChange(change)
-		}
-
-		const handleChanges = (changes: TLAiChange[]) => {
-			for (const transform of transforms) {
-				if (transform.transformChanges) {
-					changes = transform.transformChanges(changes)
-				}
-			}
+			apply({ change, editor: this.opts.editor })
 		}
 
 		return {
-			prompt: _prompt,
+			prompt,
 			handleChange,
-			handleChanges,
```

Comment:
I think we should restore `handleChanges`. Some people could have been using it. No need to remove it.